module/project/templates/helper.js

```diff
@@ -957,74 +958,190 @@ module.exports = class ProjectTemplatesHelper {
+                            status : HTTP_STATUS_CODE.bad_request.status,
+                            message : CONSTANTS.apiResponses.PROJECT_TEMPLATE_NOT_FOUND
+                        }
+                    }
+    
+                    if (templateData[0].tasks && templateData[0].tasks.length > 0) {
+                        templateData[0].tasks = 
+                        await this.tasksAndSubTasks(templateData[0]._id);
+                    }
+    
+                    let result = await _templateInformation(templateData[0])
+                    if( !result.success ) {
+                        return resolve(result);
+                    }
+    
+                    if( !templateData[0].isReusable ) {
+                        
+                        templateData[0].projectId = "";
+    
+                        let project = await projectQueries.projectDocument({
+                            userId : userId,
+                            projectTemplateId : templateData[0]._id
+                        },["_id"]);
+    
+                        if(project && project.length > 0){
+                            templateData[0].projectId = project[0]._id;
+                        }
+                    }
+    
+                    return resolve({
+                        success : false,
+                        data : result.data,
+                        message : CONSTANTS.apiResponses.PROJECT_TEMPLATE_DETAILS_FETCHED
+                    });
                 }
 
-                let result = await _templateInformation(templateData[0])
-
-                if( !result.success ) {
-                    return resolve(result);
-                }
+                //get data when link is given
+                if(link){
+                    
+                    let queryData={};
+                    queryData["link"]=link;
+
+                    let solutionDocument = 
+                    await coreService.solutionsDocuments(
+                        queryData,
+                        [
+                            "_id",
+                            "name",
+                            "programId",
+                            "programName",
+                            "externalId",
+                            "projectTemplateId",
+                            "link"
+                        ],
+                        "none"
+                        );
 
-                if( !templateData[0].isReusable ) {
+                    if( !solutionDocument.success || solutionDocument.data.length===0 ) {
+                        throw {
+                            message : CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+                            status : HTTP_STATUS_CODE['bad_request'].status
+                        }
+                    }
                     
-                    templateData[0].projectId = "";
+                    let solutiondata=solutionDocument.data;
+                    let templateId=solutiondata.map(function(id){
```

Comment:
@VISHNUDAS-tunerlabs no need to do map here .
let templateId=solutiondata[0].projectTemplateId